lib/adf_diag.py

```diff
@@ -1080,3 +1108,227 @@ def derive_variables(self, vars_to_derive=None, ts_dir=None, overwrite=None):
+
+        mdtf_model_data_root = self.get_mdtf_info("MODEL_DATA_ROOT")
+
+        # These MDTF words for day & month .But CESM will have hour_6 and hour_3, etc.
+        # Going to need a dict to translate.
+        # Use cesm_freq_strings = freq_string_options.keys
+        # and then freq = freq_string_option(freq_string_found)
+        freq_string_options = ["month", "day", "6hr", "3hr", "1hr"]
+
+        hist_str_list = self.get_cam_info("hist_str")
+        case_names = self.get_cam_info("cam_case_name", required=True)
+        var_list = self.diag_var_list
+
+        for case_idx, case_name in enumerate(case_names):
+
+            hist_str_case = hist_str_list[case_idx]
+            for hist_str in hist_str_case:
+                if verbose > 0:
+                    print(f"\t looking for {hist_str} in {cam_ts_loc[0]}")
+                for var in var_list:
+
+                    #
+                    # Source file is ADF time series file
+                    #
+                    adf_file_str = (
+                        cam_ts_loc[case_idx]
+                        + os.sep
+                        + ".".join([case_name, hist_str, var, "*"])
+                    )  # * to match timestamp: could be multiples
+                    adf_file_list = glob.glob(adf_file_str)
+
+                    if len(adf_file_list) == 1:
+                        if verbose > 1:
+                            print(f"Copying ts file: {adf_file_list} to MDTF dir")
+                    elif len(adf_file_list) > 1:
+                        if verbose > 0:
+                            print(
+                                f"WARNING: found multiple timeseries files {adf_file_list}. Continuing with best guess; suggest cleaning up multiple dates in ts dir"
+                            )
+                    else:
+                        if verbose > 0:
+                            print(
+                                f"WARNING: No files matching {case_name}.{hist_str}.{var} found in {adf_file_str}. Skipping"
+                            )
+                        continue  # skip this case/hist_str/var file
+                    adf_file = adf_file_list[0]
+
+                    # If freq is not set, it means we just started this hist_str. So check the first ADF file to find it
+                    hist_file_ds = xr.open_dataset(
+                        adf_file, decode_cf=False, decode_times=False
+                    )
+                    if "time_period_freq" in hist_file_ds.attrs:
+                        dataset_freq = hist_file_ds.attrs["time_period_freq"]
+                        if verbose > 2:
+                            print(f"time_period_freq attribute found: {dataset_freq}")
+                    else:
+                        if verbose > 0:
+                            print(
+                                f"ERROR: Necessary 'time_period_freq' attribute missing from {adf_file}"
```

Comment:
I would change this message to the following:

```suggestion
                                f"WARNING: Necessary 'time_period_freq' attribute missing from {adf_file}. Skipping file."
```